Code/Framework/AzFramework/AzFramework/Physics/Material/PhysicsMaterialManager.cpp

```diff
@@ -62,24 +62,27 @@ namespace Physics
             return materialFound;
         }
 
-        // Take a reference so we can mutate it.
-        AZ::Data::Asset<AZ::Data::AssetData> assetLocal = materialAsset;
-        if (!assetLocal.IsReady())
+        auto newMaterial = CreateMaterialInternal(id, materialAsset);
+        if (newMaterial)
         {
-            assetLocal.QueueLoad();
-            assetLocal.BlockUntilLoadComplete();
-
-            // Failed to load the asset
-            if (!assetLocal.IsReady())
-            {
-                return nullptr;
-            }
+            m_materials.emplace(newMaterial->GetId(), newMaterial);
```

Comment:
suggestion: As `newMaterial` isn't used after this it might be worth moving to an if initializer - we might also then want to `AZStd::move` the value into the `m_materials` container

```

if (auto newMaterial = CreateMaterialInternal(id, materialAsset);
    newMaterial)
{
    m_materials.emplace(newMaterial->GetId(), AZStd::move(newMaterial));
```

Reply:
`newMaterial` is returned at the end, so it can't be moved.